Description: Use local object inventory files for sphinx
 Use the local object inventory files to prevent sphinx-build from 
 downloading them during build.
 This patch makes the intersphinx mappings depend on the existence of 
 these files, thus passing Debian Policy 12.3 while still maintaining 
 a level of comfort for the reader of the documentation.
Author: Fladischer Michael <FladischerMichael@fladi.at>
Forwarded: not-needed
Last-Update: 2011-03-03

---
 doc/conf.py |   17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

--- json-schema-validator-2.3.orig/doc/conf.py
+++ json-schema-validator-2.3/doc/conf.py
@@ -196,6 +196,21 @@ latex_documents = [
 # If false, no module index is generated.
 #latex_use_modindex = True
 
 
 # Example configuration for intersphinx: refer to the Python standard library.
-intersphinx_mapping = {'http://docs.python.org/': None}
+#intersphinx_mapping = {'http://docs.python.org/': None}
+def check_object_path(key, url, path):
+    if os.path.isfile(path):
+        return {key: (url, path)}
+    return {}
+
+intersphinx_mapping = {}
+intersphinx_mapping.update(check_object_path('python',
+                                            'http://docs.python.org/',
+                                            '/usr/share/doc/python'
+                                              + '.'.join([str(x) for x in sys.version_info[0:2]])
+                                              + '/html/objects.inv'))
+intersphinx_mapping.update(check_object_path('sphinx',
+                                             'http://sphinx.pocoo.org/',
+                                             '/usr/share/doc/python-sphinx/html/objects.inv'))
+
